x-pack/plugins/enterprise_search/public/applications/shared/kibana/kibana_logic.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { ApplicationStart, ChromeBreadcrumb } from 'src/core/public';
+
+export interface IKibanaValues {
+  config: { host?: string };
+  navigateToUrl: ApplicationStart['navigateToUrl'];
+  setBreadcrumbs(crumbs: ChromeBreadcrumb[]): void;
+  setDocTitle(title: string): void;
+}
+
+export const KibanaLogic = kea<MakeLogicType<IKibanaValues>>({
+  path: ['enterprise_search', 'kibana_logic'],
+  reducers: ({ props }) => ({
```

Comment:
Nice. I didn't realize you could initialize logic with props like this.

Reply:
Yeah! https://kea.js.org/docs/guide/additional/#props

Although as a warning (I think I mentioned this a few PRs back, but repeating it just as an fyi) if you set a value default to a passed prop without any fallbacks (e.g. `someVar: [props.someVar, {}],`), if you ever mount without passing a `someVar` prop the entire Kea/redux instance will error/crash. It's a good way of making sure you always have the data you need I guess? 😄